spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/dialect/NullTypeStrategy.java

```diff
@@ -0,0 +1,33 @@
+package org.springframework.data.jdbc.core.dialect;
+
+import java.sql.JDBCType;
+import java.sql.SQLType;
+
+/**
+ * Interface for defining what to {@link SQLType} to use for {@literal null} values.
+ * 
+ * @author Jens Schauder
+ * @since 4.0
+ */
+public interface NullTypeStrategy {
+
+	/**
+	 * Implementation that always uses {@link JDBCType#NULL}. Suitable for all databases that actually support this
+	 * {@link JDBCType}.
+	 */
+	NullTypeStrategy DEFAULT = sqlType -> JDBCType.NULL;
```

Comment:
We should have more meaningful names. Something like `NULL_VALUE` instead of `DEFAULT`